packages/datadog-instrumentations/src/helpers/register.js

```diff
@@ -131,12 +132,19 @@ for (const packageName of names) {
             loadChannel.publish({ name, version, file })
             // Send the name and version of the module back to the callback because now addHook
             // takes in an array of names so by passing the name the callback will know which module name is being used
-            moduleExports = hook(moduleExports, version, name)
-            // Set the moduleExports in the hooks weakmap
-            hook[HOOK_SYMBOL].set(moduleExports, name)
+            // TODO(BridgeAR): This is only true in case the name is identical
+            // in all loads. If they deviate, the deviating name would not be
+            // picked up due to the unification. Check what modules actually use the name.
+            // TODO(BridgeAR): Only replace moduleExports if the hook returns a new value.
+            // This allows to reduce the instrumentation code (no return needed).
+            const newModuleExports = hook(moduleExports, version, name)
+            if (newModuleExports) {
+              moduleExports = newModuleExports
+            }
```

Comment:
Slight readability improvement:

```suggestion
            moduleExports = hook(moduleExports, version, name) || moduleExports
```

Of if you actually only care about `null` and `undefined`, it could be:

```suggestion
            moduleExports = hook(moduleExports, version, name) ?? moduleExports
```

Reply:
Good point!